### PR TITLE
Backport #16837 to 1.7.7.x : Add breadcrumbs to my account pages

### LINKED ISSUE #16836: Breadcrumb missing on my account page
There is no breadcrumd in my account page on the front office.

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 
Original description : 

If I make a PR fast, can we make it to 1.7.7?

**Missing breadcrumbs in these controllers**
bestsales
newproducts
pricesdrop
myaccount
password
Login page + registration

**Incomplete breadcrumbs in these controllers**
history - last key is missing
identity - last key is missing
discount - last key is missing
addresses - last key is missing
orderslip - last key is missing

**Additional information**
PrestaShop version: 1.7.6.2

### LINKED ISSUE #18122: Last breadcrumb element "Your account" is not clickable in the Your account blocks
#### Describe the bug

In the "Your account" blocks, the last breadcrumb element is not clickable.
For example in the  "Your address" page
image

#### Expected behavior

We should have `Home / Your account / Your address`

So, `Your account` is clickable

#### Steps to Reproduce

1. Go to FO 
2. Sign in
3. Go to "Your account" page
4. Check all blocks
5. The last breadcrumb element is not clickable

It is not the same case with PS1.7.6

**Screenshots**
PS1.7.6.4: https://drive.google.com/file/d/1YdUwkeHvEPqe6s4x_41mtf6XoSJR-R6j/view
PS1.7.7: https://drive.google.com/file/d/1aErDxq9Q6c7oX-bCPXOj2Y_VK2nDNvvr/view

#### Additional information

* PrestaShop version: 1.7.7.x
* PHP version: 7.2

### PR DESCRIPTION
**Pull request by matks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | Adds missing breadcrumbs to my acount, login, password and registration pages, to make navigation easier for the customer. I used the same strings from the templates.
| Type?         | improvement
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? |  https://github.com/PrestaShop/PrestaShop/issues/18122
| How to test?  | -

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/18221)

### CODE REVIEW

File: controllers/front/GuestTrackingController.php [hunk 1]
```diff
@@ -139,8 +140,9 @@ public function initContent()
         $this->context->smarty->assign([
             'order' => $presented_order,
             'guest_email' => Tools::getValue('email'),
-            'HOOK_DISPLAYORDERDETAIL' => Hook::exec('displayOrderDetail', ['order' => $this->order]),
-        ]);
+            'is_customer' => Customer::customerExists(Tools::getValue('email'), false, true),
```
Comment (matks) [inline controllers/front/GuestTrackingController.php]: Looks like backport mistake

**Review by PierreRambaud — DISMISSED** [review]

File: controllers/front/IdentityController.php [hunk 2]
```diff
@@ -82,6 +82,11 @@ public function getBreadcrumbLinks()
 
         $breadcrumb['links'][] = $this->addMyAccountToBreadcrumb();
 
+        $breadcrumb['links'][] = [
+            'title' => $this->trans('Your personal information', array(), 'Shop.Theme.Customeraccount'),
```
Comment (Arman-Hosseini) [inline controllers/front/IdentityController.php]:
```suggestion
            'title' => $this->trans('Your personal information', [], 'Shop.Theme.Customeraccount'),
```

File: controllers/front/HistoryController.php [hunk 3]
```diff
@@ -108,6 +108,11 @@ public function getBreadcrumbLinks()
 
         $breadcrumb['links'][] = $this->addMyAccountToBreadcrumb();
 
+        $breadcrumb['links'][] = [
+            'title' => $this->trans('Order history', array(), 'Shop.Theme.Customeraccount'),
```
Comment (Arman-Hosseini) [inline controllers/front/HistoryController.php]:
```suggestion
            'title' => $this->trans('Order history', [], 'Shop.Theme.Customeraccount'),
```

File: controllers/front/DiscountController.php [hunk 4]
```diff
@@ -89,6 +89,11 @@ public function getBreadcrumbLinks()
 
         $breadcrumb['links'][] = $this->addMyAccountToBreadcrumb();
 
+        $breadcrumb['links'][] = [
+            'title' => $this->trans('Your vouchers', array(), 'Shop.Theme.Customeraccount'),
```
Comment (Arman-Hosseini) [inline controllers/front/DiscountController.php]:
```suggestion
            'title' => $this->trans('Your vouchers', [], 'Shop.Theme.Customeraccount'),
```

**Review by Progi1984 — APPROVED** [review]

**Comment by Hlavtox** [comment]
@matks All is good bro! Thanks for help 🍺

**Review by matthieu-rolland — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by sarahdib** [comment]
@matks 
The last element is not clickable with the PR

**Comment by Hlavtox** [comment]
@sarahdib It's not meant to be clickable.

The last element of breadcrumbs is rendered as a span and not a link. See this merged PR.

https://github.com/PrestaShop/PrestaShop/pull/14710

**Comment by Robin-Fischer-PS** [comment]
Could we have a "How to test" @matks ? I'm a bit confused right now ^^
Since the fixed issue says last breadcrumps should be clickable, but in fact no, well, what's does this PR do ?

**Comment by Hlavtox** [comment]
@Robin-Fischer-PS It fixes it by adding the last key to the navigation. So the key, that should be clickable, is clickable.

The purpose of breacrumbs is to add a navigation to pages that are above the one you are on. There is no point of having the last key as a link. Again, as done (nice!) here: https://github.com/PrestaShop/PrestaShop/pull/14710 by @matthieu-rolland .

But, on some pages where some breacrumbs were missing, I added them.

**What already is done in Prestashop**
Products
products
Categories
category

**I went throught all my account pages and added the missing breadcrumbs everywhere.**
Order history
orders

Addresses
addresses

**Comment by sarahdib** [comment]
Hello @Hlavtox 

Thank you for this precision. It's clearer now.

A how to test clear and complete is very important in order to test correctly :)

**Comment by Progi1984** [comment]
Thanks @matks & @Hlavtox

**Pull request merged**